tools/SourceKit/lib/SwiftLang/SwiftEditor.cpp

```diff
@@ -2239,7 +2243,10 @@ void SwiftLangSupport::editorReplaceText(StringRef Name,
     }
     Snapshot = EditorDoc->replaceText(Offset, Length, Buf,
                                       Consumer.needsSemanticInfo());
-    assert(Snapshot);
+    if (!Snapshot) {
+      Consumer.handleRequestError("Failed to replace text");
```

Comment:
Since the only known failure is range error, should the error say something like `"Offset+Length is out of range"` ?
Alternatively if you want to keep things flexible you could add a `std::string &error` option for `EditorDoc->replaceText()` to provide the error string.

Reply:
Added `std::string &error` option for `EditorDoc->replaceText()`, and added error message check in the test case.